core/domain/platform_parameter_domain.py

```diff
@@ -376,29 +440,33 @@ def _match_version_expression(self, op, value, client_version):
         elif op == '>=':
             return is_equal or is_client_version_larger
 
-    def _is_first_version_smaller(self, version_a, version_b):
+    def _is_first_version_smaller(self, version_a: str, version_b: str) -> bool:
         """Compares two version strings, return True if the first version is
         smaller.
 
         Args:
-            version_a: str. The version string (e.g. '1.0.0').
-            version_b: str. The version string (e.g. '1.0.0').
+            version_a_str: str. The version string (e.g. '1.0.0').
+            version_b_str: str. The version string (e.g. '1.0.0').
 
         Returns:
             bool. True if the first version is smaller.
         """
-        version_a = version_a.split('.')
-        version_b = version_b.split('.')
+        version_a_list = version_a.split('.')
+        version_b_list = version_b.split('.')
 
         for sub_version_a, sub_version_b in python_utils.ZIP(
-                version_a, version_b):
+                version_a_list, version_a_list):
             if int(sub_version_a) < int(sub_version_b):
                 return True
             elif int(sub_version_a) > int(sub_version_b):
                 return False
         return False
 
-    def _match_version_flavor(self, op, flavor, client_version):
+    # Type ignore because if client_verison is None then method stops
+    # line 462 with a False return
+    def _match_version_flavor(
+        self, op: str, flavor: str, client_version: Optional[str]
+        ) -> bool:
```

Comment:
```suggestion
    # Type ignore because if client_verison is None then method stops
    # line 462 with a False return
    def _match_version_flavor(
        self, op: str, flavor: str, client_version: Optional[str]
    ) -> bool:
```

Where is the type ignore?

Reply:
done